lib/AST/AccessRequests.cpp

```diff
@@ -25,12 +31,183 @@ namespace swift {
 #undef SWIFT_TYPEID_HEADER
 }
 
+//----------------------------------------------------------------------------//
+// Access Scope Checkers
+//----------------------------------------------------------------------------//
+
+AccessScopeChecker::AccessScopeChecker(const DeclContext *useDC,
+                                       bool treatUsableFromInlineAsPublic)
+  : File(useDC->getParentSourceFile()),
+    TreatUsableFromInlineAsPublic(treatUsableFromInlineAsPublic),
+    Context(File->getASTContext()) {}
+
+bool
+AccessScopeChecker::visitDecl(ValueDecl *VD) {
+  if (!VD || isa<GenericTypeParamDecl>(VD))
+    return true;
+
+  // FIXME: Figure out why AssociatedTypeDecls don't always have an access
+  // level here.
+  if (!VD->hasAccess()) {
```

Comment:
This might no longer happen. If it does, ensure that associated types always get the access level from their protocol